website/src/content/docs/docs/extending-typespec/emitter-framework.md

```diff
@@ -3,369 +3,118 @@ id: emitter-framework
+There are also some additional components handy for emitting TypeScript from TypeSpec:
 
-Lets return to our previous example:
-
-```typescript
-  modelDeclarationContext(model: Model, name: string): Context {
-    if (model.name === "Person") {
-      const parentScope = this.emitter.getContext().scope;
-      const scope = this.emitter.createScope({}, "Namespace", parentScope);
-
-      return {
-        scope,
-      };
-    } else {
-      return {};
-    }
-  }
-```
+* `<ArrayExpression elementType={Type} />` - emit a TypeScript array type of the given type.
+* `<RecordExpression elementType={Type} >` - emit a TypeScript record type of the given type.
 
-We can now see how this results in the `Person` model being located in a nested scope - because we set `scope` on the context to a new scope we created via `this.emitter.setScope`.
+### Example TypeScript emitter
 
-### Extending `TypeEmitter`
+This example emitter will emit TypeScript types for some TypeSpec types.
 
-TypeEmitters are classes and explicitly support subclassing, so you can customize an existing emitter by extending it and overriding any methods you want to customize in your subclass. In fact, emitters you find out in the ecosystem are likely not to work without creating a subclass, because they only know how to emit types, but you need to provide the scope for any declarations it wants to create. For example, if we have a base `TypeScript Emitter` that can convert TypeSpec into TypeScript, we might extend it to tell it to put all declarations in the same file:
+```tsx
+import { Output, SourceDirectory, SourceFile } from "@alloy-js/core";
+import * as ts from "@alloy-js/typescript";
+import { EmitContext } from "@typespec/compiler";
+import { writeOutput } from "@typespec/emitter-framework";
 
-```typescript
-class MyTsEmitter extends TypeScriptEmitter {
-  programContext(program: Program): Context {
-    const sourceFile = this.emitter.createSourceFile("test.txt");
-    return {
-      scope: sourceFile.globalScope,
-    };
-  }
-}
-```
-
-Or, if we want one class or interface per file, we might instead do something like:
-
-```typescript
-class MyTsEmitter extends TypeScriptEmitter {
-  modelDeclarationContext(program: Program): Context {
-    const sourceFile = this.emitter.createSourceFile("test.txt");
-    return {
-      scope: sourceFile.globalScope,
-    };
-  }
-  // and similar for other declarations: Unions, Enums, Interfaces, and Operations.
+export async function $onEmit(context: EmitContext) {
+  // get some models from the type graph
+  const types = getInterestingTypes();
```

Comment:
showing also some basic example of what it looks like here calling typekits would be nice, as they are mentioned but not shown in example